index.js

```diff
@@ -252,6 +252,11 @@ function showHistory(friendName) {
 
 // Printing helpers
 function prettyPrintFriend(friend, eventsByFriend) {
+  var friends = loadFriendsData();
+  var myFriend = getFriendByName(friends, friend)[0];
+  if (!myFriend) {
+    fail("Friend " + friend + " does not exist");
```

Comment:
This gets called each time you want to display the `friend`. 

I would recommend putting this in the root `showHistory` function.